code/game/objects/items/chainsaw.dm

```diff
@@ -0,0 +1,87 @@
+
+// CHAINSAW
+/obj/item/chainsaw
+	name = "chainsaw"
+	desc = "A versatile power tool. Useful for limbing trees and delimbing humans."
+	icon_state = "chainsaw_off"
+	lefthand_file = 'icons/mob/inhands/weapons/chainsaw_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/chainsaw_righthand.dmi'
+	flags_1 = CONDUCT_1
+	force = 13
+	var/force_on = 24
+	w_class = WEIGHT_CLASS_HUGE
+	throwforce = 13
+	throw_speed = 2
+	throw_range = 4
+	custom_materials = list(/datum/material/iron=13000)
+	attack_verb = list("sawed", "torn", "cut", "chopped", "diced")
+	hitsound = "swing_hit"
+	sharpness = IS_SHARP
+	actions_types = list(/datum/action/item_action/startchainsaw)
+	tool_behaviour = TOOL_SAW
+	toolspeed = 0.5
+	var/on = FALSE
+	var/wielded = FALSE // track wielded status on item
+
+/obj/item/chainsaw/Initialize()
+	. = ..()
+	AddComponent(/datum/component/butchering, 30, 100, 0, 'sound/weapons/chainsawhit.ogg', TRUE)
+	AddComponent(/datum/component/two_handed, require_twohands=TRUE, \
+				on_wield_callback=CALLBACK(src, .proc/on_wield), on_unwield_callback=CALLBACK(src, .proc/on_unwield))
+
+/// Callback triggered on wield of two handed item
+/obj/item/chainsaw/proc/on_wield(mob/user)
+	wielded = TRUE
+
+/// Callback triggered on unwield of two handed item
+/obj/item/chainsaw/proc/on_unwield(mob/user)
+	wielded = FALSE
+
+/obj/item/chainsaw/suicide_act(mob/living/carbon/user)
+	if(on)
+		user.visible_message("<span class='suicide'>[user] begins to tear [user.p_their()] head off with [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(src, 'sound/weapons/chainsawhit.ogg', 100, TRUE)
+		var/obj/item/bodypart/head/myhead = user.get_bodypart(BODY_ZONE_HEAD)
+		if(myhead)
+			myhead.dismember()
+	else
+		user.visible_message("<span class='suicide'>[user] smashes [src] into [user.p_their()] neck, destroying [user.p_their()] esophagus! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(src, 'sound/weapons/genhit1.ogg', 100, TRUE)
+	return(BRUTELOSS)
+
+/obj/item/chainsaw/attack_self(mob/user)
+	on = !on
+	to_chat(user, "As you pull the starting cord dangling from [src], [on ? "it begins to whirr." : "the chain stops moving."]")
+	force = on ? force_on : initial(force)
+	throwforce = on ? force_on : initial(force)
+	icon_state = "chainsaw_[on ? "on" : "off"]"
+	var/datum/component/butchering/butchering = src.GetComponent(/datum/component/butchering)
+	butchering.butchering_enabled = on
+
+	if(on)
+		hitsound = 'sound/weapons/chainsawhit.ogg'
+	else
+		hitsound = "swing_hit"
+
+	if(src == user.get_active_held_item()) //update inhands
+		user.update_inv_hands()
+	for(var/X in actions)
+		var/datum/action/A = X
+		A.UpdateButtonIcon()
+
+/obj/item/chainsaw/get_dismemberment_chance()
```

Comment:
should be in the component

Reply:
Following Ninja's suggestion to track wielded status with signals this item will manage it's own logic as it needs.

Reply:
Closed per Ninja's resolve of the same issue.
https://github.com/tgstation/tgstation/pull/49422#discussion_r382632611